static/js/server_events_dispatch.js

```diff
@@ -287,11 +287,13 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
             'left_side_userlist',
             'timezone',
             'twenty_four_hour_time',
+            'translate_emoticons',
         ];
         if (_.contains(user_display_settings, event.setting_name)) {
             page_params[event.setting_name] = event.setting;
         }
-        if (event.setting_name === 'twenty_four_hour_time') {
+        if (event.setting_name === 'twenty_four_hour_time'
+            || event.setting_name === 'translate_emoticons') {
```

Comment:
This doesn't require a rerender, since it only impacts future messages you send, not old ones.  I think the default block of updating `page_params` is all we need.

Reply:
@timabbott OK, I undid this change. I think it was a holdover from a different way I was doing it before. Thanks!